frontend/packages/console-app/src/components/resource-quota/AppliedClusterResourceQuotaCharts.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { DonutChart } from '@console/internal/components/graphs/donut';
+import { AppliedClusterResourceQuotaKind } from '@console/internal/module/k8s';
+import { getUsedPercentage, getLabelAndUsage } from './utils';
+
+import './resource-quota.scss';
+
+type AppliedClusterResourceQuotaChartsProps = {
+  appliedClusterResourceQuota: AppliedClusterResourceQuotaKind;
+  namespace: string;
+};
+
+const AppliedClusterResourceQuotaCharts = ({
+  appliedClusterResourceQuota,
+  namespace,
+}: AppliedClusterResourceQuotaChartsProps): JSX.Element => {
+  const { t } = useTranslation();
+  const nsQuotas = appliedClusterResourceQuota.status?.namespaces?.find(
+    (ns) => ns.namespace === namespace,
+  );
```

Comment:
You can request `AppliedClusterResourceQuota` for a namespace from the API (as if it were a namespace scoped resource) so you don't need to filter on the client.